lnbits/extensions/scheduledpayments/views_api.py

```diff
@@ -0,0 +1,101 @@
+from http import HTTPStatus
+
+import crontools as ct
+from fastapi import Query
+from fastapi.params import Depends
+from loguru import logger
+from starlette.exceptions import HTTPException
+
+from lnbits.core.crud import get_user
+from lnbits.decorators import WalletTypeInfo, get_key_type, require_admin_key
+
+from . import scheduledpayments_ext
+from .crud import (
+    create_schedule,
+    delete_schedule,
+    get_events,
+    get_schedule,
+    get_schedules,
+    update_schedule,
+)
+from .models import CreateScheduleData, UpdateScheduleData
+
+
+@scheduledpayments_ext.get("/api/v1/schedule/{schedule_id}", status_code=HTTPStatus.OK)
+async def api_schedule(schedule_id: str):
+    schedule = await get_schedule(schedule_id)
+    if not schedule:
+        raise HTTPException(
+            status_code=HTTPStatus.NOT_FOUND, detail="Schedule does not exist."
+        )
+
+    return schedule.dict()
+
+
+@scheduledpayments_ext.get("/api/v1/schedule", status_code=HTTPStatus.OK)
+async def api_schedules(
+    all_wallets: bool = Query(None), wallet: WalletTypeInfo = Depends(get_key_type)
+):
+    wallet_ids = [wallet.wallet.id]
+    if all_wallets:
+        wallet_ids = (await get_user(wallet.wallet.user)).wallet_ids
+
+    return [schedule.dict() for schedule in await get_schedules(wallet_ids)]
+
+
+@scheduledpayments_ext.post("/api/v1/schedule", status_code=HTTPStatus.CREATED)
+async def api_schedule_create(
+    data: CreateScheduleData, wallet: WalletTypeInfo = Depends(get_key_type)
```

Comment:
Usually `create` should require `admin_key`:
`wallet: WalletTypeInfo = Depends(require_admin_key)`

Reply:
- same for the other `C_UD` endpoints

Reply:
Fix pushed.